SEED-platform/seed
# fix: resolve several member permission issues

#### Any background context you want to provide?
See the related issue

#### What's this PR do?
Improves the logic around whether someone can edit org member roles

#### How should this be manually tested?
- Check that removing a user with only one org doesn't show the modal about deleting the user
- Check that roles are now in hierarchical order (owner -> member -> viewer)
- Check that the following revised rules work as expected:
    - You must be either an `owner` or `member` to see an org's members page (this hasn't changed)
    - Superusers can edit any user's role except for the org owner if there is only one
    - Owners can edit all roles except their own if they are the only owner
    - Members can only edit their own role, and only to change it to "viewer"
    - Non-superusers cannot promote their role
    - Once you demote yourself you no longer see options to promote yourself and the UI reflects your new permissions

#### What are the relevant tickets?
#2677
